lib/dark_worlds_server/matchmaking/matching_coordinator.ex

```diff
@@ -40,10 +40,14 @@ defmodule DarkWorldsServer.Matchmaking.MatchingCoordinator do
     {:reply, :ok, %{state | players: players, session: session_ref}}
   end
 
-  def handle_call({:join, user_id}, {from, _}, state) do
-    players = state.players ++ [{user_id, from}]
-    send(self(), :check_capacity)
-    {:reply, :ok, %{state | players: players}}
+  def handle_call({:join, user_id}, {from, _}, %{players: players} = state) do
+    if Enum.any?(players, fn {player_user_id, _} -> player_user_id == user_id end) do
+      {:reply, :ok, state}
+    else
+      players = state.players ++ [{user_id, from}]
```

Comment:
If we don't care about order use instead:
```suggestion
      players = [{user_id, from}| players]
```
Check [this](https://www.erlang.org/doc/efficiency_guide/myths#myth--operator--++--is-always-bad) on why.

Reply:
This is not only an upgrade, it also fix the numbers of the players in the match, with this change the first player to enter the queue is the numer 1 now instead of 10, nice!

Reply:
An upgrade indeed 🚀